homeassistant/components/vera/__init__.py

```diff
@@ -80,8 +93,9 @@ def stop_subscription(event):
 
     # Initialize the Vera controller.
     controller, _ = veraApi.init_controller(base_url)
-    hass.data[VERA_CONTROLLER] = controller
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_subscription)
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, lambda event: controller.stop()
+    )
 
     try:
         all_devices = controller.get_devices()
```

Comment:
I/O must be scheduled on the executor thread pool with `hass.async_add_executor_job`.